### PR TITLE
Add KLU feature

### LINKED ISSUE #3: Sundials KLU linear solver
I'm interested in using the Sundials KLU linear solver (https://sundials.readthedocs.io/en/latest/sunlinsol/SUNLinSol_links.html#the-sunlinsol-klu-module). I had a look and couldn't see a feature to enable this, so I assume this has not been implemented? Note that  it would require linking against suitesparse (https://github.com/DrTimothyAldenDavis/SuiteSparse)

I would be happy to work on this issue and make a PR

### PR DESCRIPTION
**Pull request by martinjrobins**
Fixes #3. This PR:

- adds `klu` feature
- adds `SuiteSparse` submodule (https://github.com/DrTimothyAldenDavis/SuiteSparse)
- if `klu` feature is added, then bindings are added for the sundials KLU linear solver (requires both sundials and suitesparse to be installed)
- If a vendor build, also builds and links suitesparse (using the `SuiteSparse` submodule) to the sundials library build (requires a BLAS installation to be present)

Note: I havn't added the klu feature to the CI yet. If would be good to get some feedback on how this should be tested. Should klu be a default feature, or should this be tested separately? Also, the current CI has an issue that I noticed: https://github.com/Chris00/sundials-sys/issues/5

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -150,18 +250,21 @@ fn main() {
 
     // Second, we use bindgen to generate the Rust types
 
-    let bindings_rs = PathBuf::from(env::var("OUT_DIR").unwrap())
-        .join("bindings.rs");
+    let bindings_rs = PathBuf::from(env::var("OUT_DIR").unwrap()).join("bindings.rs");
```
Comment (Chris00) [inline build.rs]: May you please refrain from such cosmetic changes which add noise to the diff?  I understand that one may want to run `rustfmt` on the file but then it is better IMHO to make it a separate PR.

Comment (martinjrobins) [reply build.rs]: apologies, I typically have autoformat on in my IDE, so I didn't realise I was making these changes.

Comment (martinjrobins) [reply build.rs]: removed now

**Comment by Chris00** [comment]
Thanks for your PR.  I builds fine with the system Sundials.  I see two issues we need to resolve.
- I think the paths for local build are not correct (`cargo build --features klu` fails) and one should act an extra environment variable for SuiteSparse — the default should be assuming it is installed _alongside_ Sundials (`$D/sundials/` and `$D/suitesparse/`)  and not as a sub-directory.  I can fix these but these paths may change due to the second issue.
- SuiteSparse repository is 268Mb while a crate can [at max be 10Mb](https://doc.rust-lang.org/cargo/reference/publishing.html#packaging-a-crate) — and most of the 10Mb are already taken by Sundials (stripped of its doc, examples,...).  So suitesparse should be dynamically downloaded (say with [gitoxide](https://crates.io/crates/gitoxide) but an additional large download may be frowned upon, especially because it will increase the size of each `target/`) or one must require the user to first install SuiteSparse and point to its installation with an environment variable.

**Comment by Chris00** [comment]
For the first point, I meant with the vendor libraries; `cargo build --features klu,build_libraries` to force it.

**Comment by martinjrobins** [comment]
regarding the 2nd point: you're right, I didn't think about the size of the submodule. I'm happy with requiring the user to have a local installation of suitesparse and pointing to it via an environment variable. I think this is a reasonable requirement for an optional feature. I can make this change to the PR if you're also happy with this?

**Comment by Chris00** [comment]
I pushed [a commit](https://github.com/Chris00/sundials-sys/commit/2e293e92173a011bedf56014c8349888fded4c12) enabling the KLU feature (in the `dev` branch).  May you check that it works for you?  If so, I'll make a new release.

**Comment by martinjrobins** [comment]
The `dev` branch works perfectly for me, thanks @Chris00!

**Comment by Chris00** [comment]
Thanks.  I have released [version 0.5](https://crates.io/crates/sundials-sys/0.5.0).
